app/models/contact.rb

```diff
@@ -27,4 +29,18 @@ def format_address(separator)
      .reject(&:empty?)
      .join(separator)
   end
+
+  def validate_contact_type
+    address_type_codes = CategoryReference.list_by_category(:contact_type).map do |category| 
```

Comment:
Not sure whether we should make the name consist each  for address type  or contact type?